### PR TITLE
VS-12468 ReD params using CRITERION tag

### PR DESCRIPTION
**Pull request by artur-dev**

### CODE REVIEW

File: src/Message/BaseRequest.php [hunk 1]
```diff
@@ -118,6 +118,13 @@ private function getHeaders()
         return ['Content-Type' => 'application/x-www-form-urlencoded; charset=UTF-8'];
     }
 
+    public function filterEmptyValues(array $array)
+    {
+        return array_filter($array, function ($v) {
+            return $v !== null && $v !== '' && $v !== false;
```
Comment (eugene-chernogradski) [inline src/Message/BaseRequest.php]: Wrapping the boole conditions to parenthesis makes it a bit more readable:
```php
return ($v !== null) && ($v !== '') && ($v !== false);
```
And wouldn't
```php
return (string)$v === '';
```
do the same?

Comment (artur-dev) [inline src/Message/BaseRequest.php]: Parenthesis will be useful. About typecasting to string - I will check behaviour with other cases.

File: src/Message/PaymentRequest.php [hunk 2]
```diff
@@ -264,22 +264,30 @@ protected function getRedParams($card)
...
+            'CRITERION.RED_SHIP_LNAME' => $card->getShippingLastName(),
+            'CRITERION.RED_SHIP_HOME_PHONE' => $card->getShippingPhone(),
+            'CRITERION.RED_SHIP_MOBILE_PHONE' => $card->getShippingFax(),
+            'CRITERION.RED_SHIP_POSTAL_CD' => $card->getShippingPostcode(),
+            'CRITERION.RED_SHIP_STPR_CD' => $card->getShippingState(),
         ];
+        foreach (($this->getItemAmount() ?: []) as $idx => $value) {
+            $params["CRITERION.RED_ITEM_AMT$idx"] = $value;
+        }
+        foreach (($this->getItemQty() ?: []) as $idx => $value) {
```
Comment (eugene-matvejev) [inline src/Message/PaymentRequest.php]: :+1:  i will memorize this :)

File: src/Message/BaseRequest.php (hunk 1 above)
Comment (eugene-matvejev) [inline src/Message/BaseRequest.php]: I like @eugene-chernogradski 2nd suggestion, but I'm worry that someone who is not quite exprienced e.g. junior guys/QA/front-end guys might not understand what happen there

Comment (artur-dev) [inline src/Message/BaseRequest.php]: Typecasting to string would work, but for clarity I think I will leave simple checks.

**Pull request merged**